src/engine/client/client.cpp

```diff
@@ -609,43 +609,33 @@ void CClient::SnapInvalidateItem(int SnapID, int Index)
 	}
 }
 
+static const int* SnapSearchKey(const int* pKeysStart, const int* pKeysEnd, int Key)
+{
+	int MiddleIndex = (pKeysEnd - pKeysStart) / 2;
+	int Middle = pKeysStart[MiddleIndex];
+	if(MiddleIndex == 0)
+		return Middle == Key ? pKeysStart + MiddleIndex : 0x0;
```

Comment:
Why do you need +MiddleIndex if it's 0?

Reply:
Not needed, probably just a c&p from line 623

Reply:
Good catch.